README.md

```diff
@@ -1,178 +1,317 @@
+
+To run specs against [LibSass][], the C++ Sass implementation that's used for
+[Node Sass][] and other languages' Sass wrappers, you'll need to be able to
+[build LibSass][]. Once you have all the build dependencies:
+
+[LibSass]: https://sass-lang.com/libsass
+[Node Sass]: https://npmjs.com/package/node-sass
+[build LibSass]: https://github.com/sass/libsass/blob/master/docs/build.md
+
+```sh
+# If you already have a clone of the LibSass repo, you can use that instead.
+git clone https://github.com/sass/libsass
+(cd libsass; ./script/bootstrap; make sassc)
+export SASSC_PATH=`pwd`/libsass/sassc/bin/sassc
+
+bundle exec ./sass-spec.rb --impl libsass -c $SASSC_PATH
+```
+
+### Ruby Sass
+
+Unlike other implementations, Ruby Sass is installed via Bundler and used by
+default. All you need to do to run specs against it is:
+
+```sh
+bundle exec ./sass-spec.rb
+```
+
+If you want to run specs against your own clone of Ruby Sass, edit the `Gemfile`
+and replace the `gem "sass"` line with:
 
 ```ruby
-gem 'sass', :path => "/path/to/sass/directory"
+gem "sass", :path => "path/to/ruby/sass"
 ```
 
-Then run `bundle install` and `bundle exec sass-spec.rb` will run
-against your development version of ruby sass.
+## Spec Structure
+
+Each spec is defined by a directory with an `input.scss` or `input.sass` file
+and either:
+
+* An `output.css` file, in which case the spec asserts that the Sass
+  implementation compiles the input to the output. These specs are known as
+  "success specs".
+* An `error` file, in which case the spec asserts that the Sass implementation
+  prints the error message to standard error and exits with a non-zero status
+  code when it compiles the input. These specs are known as "error specs".
+
+The path to the spec serves as the spec's name, which should tersely describe
+what it's testing. Additional explanation, if necessary, is included in a silent
+comment in the input file. Specs may also contain additional files that are used
+by the input file, as well as various other features which are detailed below.
+
+### HRX
+
+Most specs are stored in [HRX files][], which are human-readable plain-text
+archives that define a virtual filesystem. This format makes it easy for code
+reviewers to see the context of specs they're reviewing. The spec runner treats
+each HRX file as a directory with the same name as the file, minus `.hrx`. For
+example:
+
+[HRX files]: https://github.com/google/hrx#human-readable-archive-hrx
+
+```hrx
+<===> input.scss
+ul {
+  margin-left: 1em;
+  li {
+    list-style-type: none;
+  }
+}
+
+<===> output.css
+ul {
+  margin-left: 1em;
+}
+ul li {
+  list-style-type: none;
+}
+```
 
-Conversely, if you edit the Sass `Gemfile` and set `gem 'sass-spec',
-:path => "..."` then the Sass unit tests will run against your
-development version of sass-spec.
+HRX archives can contain directories, which means that they can also a single
+file can define multiple different This allows us to write many specs for the
```

Comment:
Unfinished sentence.

Reply:
Wow, this whole paragraph is a mess. Done.